planet/planet_adaptor/__main__.py

```diff
@@ -0,0 +1,245 @@
+        "workspace": workspace,
+        "bucket_name": bucket,
+        "added_keys": [],
+        "updated_keys": [key],
+        "deleted_keys": [],
+        "source": workspace,
+        "target": f"user-datasets/{workspace}",
+    }
+    logging.info(f"Sending message to pulsar: {output_data}")
+    pulsar_client = PulsarClient("pulsar://pulsar-broker.pulsar:6650")
+    producer = pulsar_client.create_producer(
+        topic="harvested",
+        producer_name=f"planet-adaptor-{workspace}-{file_id}",
+        chunking_enabled=True,
+    )
+    producer.send((json.dumps(output_data)).encode("utf-8"))
+
+
+def update_stac_item_success(
+    bucket: str, key: str, parent_folder: str, item_id: str, workspaces_domain: str
+):
+    """Update the STAC item with the assets and success order status"""
+    stac_item = retrieve_stac_item(bucket, key)
+    # List files and folders in the specified folder
+    folder_prefix = f"{parent_folder}/{item_id}/"
+    folder_objects = list_objects_in_folder(bucket, folder_prefix)
+
+    # Add all listed objects as assets to the STAC item
+    if "Contents" in folder_objects:
+        for obj in folder_objects["Contents"]:
+            file_key = obj["Key"]
+
+            # Skip if the file_key is a folder
+            if file_key.endswith("/"):
+                continue
+
+            asset_name = file_key.split("/")[-1]
+
+            # Determine the MIME type of the file
+            mime_type, _ = mimetypes.guess_type(file_key)
+            if mime_type is None:
+                mime_type = "application/octet-stream"  # Default MIME type
+
+            # Add asset link to the file
+            parts = file_key.split("/", 1)
+            workspace = parts[0]
+            file_subpath = parts[1]
+            stac_item["assets"][asset_name] = {
+                "href": f"https://{workspace}.{workspaces_domain}/files/{bucket}/{file_subpath}",
+                "type": mime_type,
+            }
+    # Mark the order as succeeded and upload the updated STAC item
+    update_stac_order_status(stac_item, item_id, OrderStatus.SUCCEEDED.value)
+    upload_stac_item(bucket, key, stac_item)
+    send_pulsar_message(bucket, key)
+
+    # Create local record of the order, to be used as the workflow output
+    write_stac_item_and_catalog(stac_item, key.split("/")[-1], item_id)
+
+
+def update_stac_item_failure(bucket: str, key: str, item_id: str):
+    """Update the STAC item with the failure order status"""
+    stac_item = retrieve_stac_item(bucket, key)
+
+    # Mark the order as failed and upload the updated STAC item
+    update_stac_order_status(stac_item, item_id, OrderStatus.FAILED.value)
+    upload_stac_item(bucket, key, stac_item)
+    send_pulsar_message(bucket, key)
+
+    # Create local record of attempted order, to be used as the workflow output
+    write_stac_item_and_catalog(stac_item, key.split("/")[-1], item_id)
+
+
+def generate_request(item_id, collection, credentials, bucket, region="eu-west-2"):
```

Comment:
Not used / needs docstring if we want to keep it for possible future use

Reply:
@hcollingwood please confirm if we do want to keep this function or if it can be deleted